src/Navigation/INavigationEngine.ts

```diff
@@ -0,0 +1,235 @@
+
+    /**
+     * Get a navigation mesh constrained position, within a particular radius
+     * @param position world position
+     * @param maxRadius the maximum distance to the constrained world position
+     * @returns the closest point to position constrained by the navigation mesh
+     */
+    getRandomPointAround(position: Vector3, maxRadius: number): Vector3;
+
+    ///
+    isSupported(): boolean;
+
+    /**
+     * Create a new Crowd so you can add agents
+     * @param maxAgents the maximum agent count in the crowd
+     * @param maxAgentRadius the maximum radius an agent can have
+     * @param scene to attach the crowd to
+     * @returns the crowd you can add agents to
+     */
+    createCrowd(maxAgents: number, maxAgentRadius: number, scene: Scene): ICrowd;
+
+    /**
+     * Release all resources
+     */
+    dispose(): void;
+}
+
+/**
+ * Crowd Interface. A Crowd is a collection of moving agents constrained by a navigation mesh
+ */
+export interface ICrowd {
+    /**
+     * Add a new agent to the crowd with the specified parameter a corresponding transformNode.
+     * You can attach anything to that node. The node position is updated in the scene update tick.
+     * @param pos world position that will be constrained by the navigation mesh
+     * @param parameters agent parameters
+     * @param transform hooked to the agent that will be update by the scene
+     * @returns agent index
+     */
+    addAgent(pos: Vector3, parameters: AgentParameters, transform: TransformNode): number;
+
+    /**
+     * Returns the agent position in world space
+     * @param index agent index returned by addAgent
+     * @returns world space position
+     */
+    getAgentPosition(index: number): Vector3;
+
+    /**
+     * Gets the agent velocity in world space
+     * @param index agent index returned by addAgent
+     * @returns world space velocity
+     */
+    getAgentVelocity(index: number): Vector3;
+
+    /**
+     * remove a particular agent previously created
+     * @param index agent index returned by addAgent
+     */
+    removeAgent(index: number): void;
+
+    /**
+     * get the list of all agents attached to this crowd
+     * @returns list of agent indices
+     */
+    getAgents() : number[];
+
+    /**
+     * Tick update done by the Scene. Agent position/velocity/acceleration is updated by this function
+     * @param deltaTime in seconds
+     */
+    update(deltaTime: number): void;
+
+    /**
+     * Asks a particular agent to go to a destination. That destination is constrained by the navigation mesh
+     * @param index agent index returned by addAgent
+     * @param destination targeted world position
+     * @returns the closest point to position constrained by the navigation mesh
+     */
+    agentGoto(index: number, destination: Vector3): void;
+
+    /**
+     * Release all resources
+     */
+    dispose() : void;
+}
+
+/**
+ * Configures an agent
+ */
+export class AgentParameters {

```

Comment:
Should be an interface

Reply:
If it becomes an interface, will I be able to create it in the playground like
`var navmeshParameters = new BABYLON.NavMeshParameters();` ?